### DISCUSSION TITLE
Errore in creazione DDT

### DISCUSSION THREAD

**Discussion by cleoni** [fattureincloud/api#42] [Q&A] [answered]
Sto lavorando sulla trasmissione di un DDT via PHP e sto usando come base il codice usato per la trasmissione ordine, che funziona.

Sulla documentazione API non è chiarissimo ma immagino che il document type da usare sia DELIVERY_NOTE

L'API mi ritorna quanto segue

[422] Client error: `POST https://api-v2.fattureincloud.it/c/#COMPANY_ID#/issued_documents` resulted in a `422 Unprocessable Entity` response:
{"error":{"message":"Invalid request.","validation_result":{"entity.name":["The entity.name field is required."]}}}

Ispezionando ciò che è stato spedito, l'unico [name] che sto passando senza valore è quello relativo al mezzo di pagamento.
E' quello il motivo dell'errore? Perché il messaggio non fa capire granché e nella documentazione non trovo nulla di specifico circa cosa sia necessario o meno quando si invia un DDT. Ecco la parte del payment_method che spedisco:

	[payment_method] => FattureInCloud\Model\PaymentMethod Object
		(
			[container:protected] => Array
				(
					[id] => 
					[name] => 
					[type] => 
					[is_default] => 
					[default_payment_account] => 
					[details] => Array
						(
							[0] => FattureInCloud\Model\PaymentMethodDetails Object
								(
									[container:protected] => Array
										(
											[title] => 
											[description] => 
										)

								)

							[1] => FattureInCloud\Model\PaymentMethodDetails Object
								(
									[container:protected] => Array
										(
											[title] => 
											[description] => 
										)

								)

							[2] => FattureInCloud\Model\PaymentMethodDetails Object
								(
									[container:protected] => Array
										(
											[title] => 
											[description] => 
										)

								)

							[3] => FattureInCloud\Model\PaymentMethodDetails Object
								(
									[container:protected] => Array
										(
											[title] => 
											[description] => 
										)

								)

							[4] => FattureInCloud\Model\PaymentMethodDetails Object
								(
									[container:protected] => Array
										(
											[title] => 
											[description] => 
										)

								)

						)

					[bank_iban] => 
					[bank_name] => 
					[bank_beneficiary] => 
					[ei_payment_method] => 
				)

		)

**Comment by valmoz** [comment, marked as answer]
Buongiorno,
come dice l'errore, il problema sta nel campo **entity.name**.
Per favore, mi passi la richiesta completa, possibilmente in formato JSON. 

Mauro

**Reply by cleoni** [reply]
Grazie, adesso ho visto. 
purtroppo non si trova nella documentazione una lista di quali campi per un DDT siano attesi o necessari, e andare avanti cercando di aggiustare il tiro basandosi sull'interpretazione dei messaggi di errore è molto faticoso.
